### DISCUSSION TITLE
[Plugin Request] Export default values of input types?

### DISCUSSION THREAD

**Discussion by borekb** [dotansimha/graphql-code-generator#3108] [General]
I have an input type that has default values expressed in the schema:

```graphql
input PaginationInput {
  first: Int = 10
  after: Int = 0
}
```

Do you think it would be possible to somehow make the values `10` and `0` be part of the generated code? Currently, I have a duplication between my schema and my code which I'd like to avoid.

I'm aware that currently, only _types_ are exported from the generated TypeScript file while this would be something different. I imagine my code would look like this:

```ts
import { defaultValues } from './generated';

defaultValues.PaginationInput.first; // 10
defaultValues.PaginationInput.after; // 0
```

**Comment by dotansimha** [comment]
Sounds like it could be a new plugin :) PRs are welcome!

**Comment by martdavidson** [comment]
This would be a very nice feature, subscribing.
